### ISSUE TITLE
Trigger an error when using a non-final key

### ISSUE THREAD

**Issue by fbnlsr** [intlify/eslint-plugin-vue-i18n#568]
**Please describe what the rule should do:**

`no-nonfinal-keys`

The rule should trigger an error when the key used for translation is not final.

At the moment, if we ask for a translation on a key that is not final (meaning it contains child keys), no error is triggered. This is particularly problematic on large code bases where people might change the dictionary key layout.

**What category should the rule belong to?**

- [ ] Enforces code style (layout)
- [x] Warns about a potential error (problem)
- [ ] Suggests an alternate way of doing something (suggestion)
- [ ] Other (please specify:)

**Example:**

```vue
<template>
  <button @click="submit">{{ t('app.save') }}
</template>
```

```json
{
  "app": {
    "save": {
      "label": "Save"
    }
  }
}
```

`eslint-plugin-vue-i18n` should trigger an error as `app.save` is not a final key.
